.github/workflows/ci.yml

```diff
@@ -0,0 +1,45 @@
+name: Continuous Integration
+
+on:
+  push:
+
+jobs:
+  test:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 1.19
+
+      - name: Run tests
+        run: go test -v ./...
+
+  build-and-publish:
+    name: Build and Publish
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ vars.DOCKER_ORG }}/awsops
```

Comment:
I prefer to have repo-specific values in variables. You could also try using a github-provided value here. I know there was some issue with `github.event.repository.name` but I have also seen it work in [other places](https://github.com/search?q=org%3Asilinternational+language%3Ayaml+repository.name&type=code).